File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteDuplicates(struct ListNode* head) {
    struct ListNode* left=head;
    struct ListNode* right=head;
    while (right!=NULL){
        while(right && left->val==right->val)
            right=right->next;
        left->next=right;
        left=right;
    }
    return head;
    
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if head is None or head.next is None : return head
        value = head.val
        prev = head
        temp = head.next

        while temp :
            if temp.val == prev.val :
                prev.next = temp.next 
            else :
                prev = temp
            temp = temp.next
        return head


        